### PR TITLE
Fixes: #8244

### PR DESCRIPTION
**Pull request by sdake**
Dockerfiles have hardcoded proxy versions.  This small PR
changes the hardcoded version to 1.0.2 for the next release of Istio.

### CODE REVIEW

File: pilot/docker/Dockerfile.proxytproxy [hunk 1]
```diff
@@ -3,7 +3,7 @@ ARG proxy_version
 ARG istio_version
 
 # Environment variables indicating this proxy's version/capabilities as opaque string
-ENV ISTIO_META_ISTIO_PROXY_VERSION "1.0.0"
```
Comment (costinm) [inline pilot/docker/Dockerfile.proxytproxy]: Can you use a $istio_version - and set it like we set the SHA ?

Comment (sdake) [reply pilot/docker/Dockerfile.proxytproxy]: @costinm I thought of that, however, during an upgrade (from `1.0.1` to `1.0.2`) `$istio_version` may show `1.0.2` while the dataplane may still be running 1.0.1 proxies.

The purpose of this env variable is for `istioctl proxy-status` to display the version of a proxy.  I think we can expect a split setup of proxy versions in some cases.

**Pull request merged**